aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/pipe/AmountOfPagesPipe.java

```diff
@@ -0,0 +1,76 @@
+/*
+   Copyright 2019, 2020 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.extensions.aspose.pipe;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.IPipeLineSession;
+import nl.nn.adapterframework.core.PipeRunException;
+import nl.nn.adapterframework.core.PipeRunResult;
+import nl.nn.adapterframework.doc.IbisDoc;
+import nl.nn.adapterframework.pipes.FixedForwardPipe;
+import nl.nn.adapterframework.stream.Message;
+import com.aspose.pdf.Document;
+
+/**
+ * Returns the amount of pages of a PDF file.
+ * 
+ * 
+ * @author Laurens Mäkel
+ * @since  7.6
+ *
+ */
+public class AmountOfPagesPipe extends FixedForwardPipe {
+	private String charset = "ISO-8859-1";
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message input, IPipeLineSession session) throws PipeRunException {
+		int result = 0;
+
+		InputStream binaryInputStream;
+		try {
+			binaryInputStream = input.asInputStream(getCharset());
+		} catch (IOException e1) {
+			throw new PipeRunException(this,
+						getLogPrefix(session) + "cannot encode message using charset [" + getCharset() + "]", e1);
+		}
+
+		try {
+			Document doc = new Document(binaryInputStream);
```

Comment:
je moet de inputstream ook sluiten. (of doet document dit!??) Ik stel voor, om voor de zekerheid, een [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) block te gebruiken.

Reply:
Done

Reply:
Het kan nog eenvoudiger. Je kan regels 53 t/m 61 vervangen door:
```
try (InputStream binaryInputStream = input.asInputStream(getCharset())) {
```
en dan regels 66 t/m 75 vervangen door een simpele catch clause